File: Exerc5/MidHandler/src/message/MessageBody.java
package message;

import java.io.Serializable;

public class MessageBody implements Serializable{
	/**
	 * 
	 */
	private static final long serialVersionUID = -1694389555503409000L;
	private RequestHeader requestHeader;
	private RequestBody requestBody;
	private ReplyHeader replyHeader;
	private ReplyBody replyBody;
	
	
	/**
	 * @param requestHeader
	 * @param requestBody
	 * @param replyHeader
	 * @param replyBody
	 */
	public MessageBody(RequestHeader requestHeader, RequestBody requestBody, ReplyHeader replyHeader,
			ReplyBody replyBody) {
		super();
		this.requestHeader = requestHeader;
		this.requestBody = requestBody;
		this.replyHeader = replyHeader;
		this.replyBody = replyBody;
	}
	/**
	 * @return the requestHeader
	 */
	public RequestHeader getRequestHeader() {
		return requestHeader;
	}
	/**
	 * @param requestHeader the requestHeader to set
	 */
	public void setRequestHeader(RequestHeader requestHeader) {
		this.requestHeader = requestHeader;
	}
	/**
	 * @return the requestBody
	 */
	public RequestBody getRequestBody() {
		return requestBody;
	}
	/**
	 * @param requestBody the requestBody to set
	 */
	public void setRequestBody(RequestBody requestBody) {
		this.requestBody = requestBody;
	}
	/**
	 * @return the replyHeader
	 */
	public ReplyHeader getReplyHeader() {
		return replyHeader;
	}
	/**
	 * @param replyHeader the replyHeader to set
	 */
	public void setReplyHeader(ReplyHeader replyHeader) {
		this.replyHeader = replyHeader;
	}
	/**
	 * @return the replyBody
	 */
	public ReplyBody getReplyBody() {
		return replyBody;
	}
	/**
	 * @param replyBody the replyBody to set
	 */
	public void setReplyBody(ReplyBody replyBody) {
		this.replyBody = replyBody;
	}
}

File: Projeto/PythonServer/message/MessageBody.py
import message_pb2

class MessageBody(object):
    #--------------Constructor--------------
    def __init__(self,requestHeader,requestBody,replyHeader,replyBody):
        self.__messageBody = message_pb2.MessageBody()
        self.__messageBody.requestHeader = requestHeader
        self.__messageBody.requestBody = requestBody
        self.__messageBody.replyHeader = replyHeader
        self.__messageBody.replyBody = replyBody

    #---------------------------------------

    #----------Getters and Setters----------
    def getRequestHeader(self):
        return self.__messageBody.requestHeader

    def getRequestBody(self):
        return self.__messageBody.requestBody

    def getReplyHeader(self):
        return self.__messageBody.replyHeader

    def getReplyBody(self):
        return self.__messageBody.replyBody

    def setRequestHeader(self,requestHeader):
        self.__messageBody.requestHeader = requestHeader

    def setRequestBody(self,requestBody):
        self.__messageBody.requestBody = requestBody

    def setReplyHeader(self,replyHeader):
        self.__messageBody.replyHeader = replyHeader

    def setReplyBody(self,replyBody):
        self.__messageBody.replyBody = replyBody

    #---------------------------------------